src/util/InteractiveMenu/InteractiveMenu.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    ActionRowBuilder,
+    ButtonBuilder,
+    ComponentType,
+    EmbedBuilder,
+    Message,
+    RepliableInteraction,
+} from "discord.js";
+import { Player } from "../Player/Player.js";
+
+export default abstract class InteractiveMenu {
+    protected message?: Message;
+    protected interaction: RepliableInteraction;
+
+    /**
+     * Creates a new PlayerMenu
+     * @param interaction This is the interaction from the command to be able to reply
+     * @param player This is a player object to be able to access stuff within player.
+     */
+    constructor(interaction: RepliableInteraction) {
+        this.interaction = interaction;
+    }
+
+    /**
+     * This will create embeds that will be displayed on the user side
+     * @returns embeds for generateMessagePayload
+     */
+    protected abstract generateEmbeds(): [EmbedBuilder];
```

Comment:
The type syntax you provided is a "tuple" of exactly 1 EmbedBuilder inside an array of exactly size 1 at index 0.
To define an array of EmbedBuilders, the following syntax should be used:
`EmbedBuilder[]` or `Array<EmbedBuilder>`.

The given is tuple syntax, which defines that the array should only have a fixed amount of items with the given types.